lib/plugins/aws/provider.js

```diff
@@ -1947,6 +1947,7 @@ class AwsProvider {
 
   getStageSourceValue() {
     const values = this.getValues(this, [
+      ['serverless', 'service', 'provider', 'apiGateway', 'stage'],
```

Comment:
This is not valid. as this function resolves service stage and not API Gateway stage, which is a different thing

Reply:
I think maybe there will be some other fields that derive their string values from the name of the api gateway stage, such as the logfile name for example. This could be quite a large PR in which case. Could you please advise on what additional acceptance criteria may need to be added to the ticket to ensure this fix is fully implemented?